packages/@n8n/api-types/src/dto/user/password-update-request.dto.ts

```diff
@@ -0,0 +1,8 @@
+import { z } from 'zod';
+import { Z } from 'zod-class';
+
+export class PasswordUpdateRequestDto extends Z.class({
+	currentPassword: z.string(),
+	newPassword: z.string(),
+	mfaCode: z.string().optional(),
+}) {}
```

Comment:
When the DTO idea was first suggested, I expected we wanted be less coupled to the validation lib. But maybe that'd be more work than we want to take on right now.

Reply:
For the auto-injection to work via decorators, the only thing that matters is that these are classes. We are not coupled to zod here, and can very quickly switch over to any other validation lib, if needed.